src/commands/deploy/deploy.ts

```diff
@@ -21,7 +22,15 @@ import { notifyDeployComplete } from './notifyDeployComplete';
 import { runPreDeployTask } from './runPreDeployTask';
 import { verifyAppSettings } from './verifyAppSettings';
 
-export async function deploy(context: IActionContext, target?: vscode.Uri | string | SlotTreeItemBase, functionAppId?: string | {}): Promise<void> {
+export async function deployProductionSlot(context: IActionContext, target?: vscode.Uri | string | SlotTreeItemBase, functionAppId?: string | {}): Promise<void> {
+    await deploy(context, target, functionAppId, ProductionSlotTreeItem.contextValue);
+}
+
+export async function deploySlot(context: IActionContext, target?: vscode.Uri | string | SlotTreeItemBase, functionAppId?: string | {}): Promise<void> {
+    await deploy(context, target, functionAppId, SlotTreeItem.contextValue);
```

Comment:
I guess if they select an app that doesn't permissions for a slot, it'll just say no resources found?  I assume this would be a small use case since most people will be on Elastic anyway, but am wondering if App Service should do this too to support command palette for slots.

Reply:
Correct. I'll try to improve the error message in a separate PR